### ISSUE TITLE
Replicating Frechet Audio Distance score for DDX7

### ISSUE THREAD

**Issue by gudgud96** [fcaspe/ddx7#1]
Hi @fcaspe,

Thanks for the wonderful work on DDX7! I am trying to replicate the Frechet Audio Distance score for the violin corpus, but I find that the FAD score I get is different from the results on the paper.

My steps are as follows:
1. Follow `Dataset Generation` to get URMP processed, so locally I have `dataset/train/violin/*.wav` and `data/train/violin/16000.h5`
2. `python train.py` using default `config.yaml` and model `tcnres_f0ld_fmstr`. I am able to see checkpoints generated under `runs/exp_test/testrun`
3. `python train.py` again by setting `mode: test` in `config.yaml`. I get `ref_test.wav` and `synth_test.wav`.
4.  For FAD calculation I am using this repo: https://github.com/gudgud96/frechet-audio-distance, based on a Torch VGGish and a pytorch-fid implementation. 
    - for Original Test Set, I calculate distance for `dataset/train/violin/*.wav` and `ref_test.wav`
    - for Synthesized Test Set, I calculate distance for `dataset/train/violin/*.wav` and `synth_test.wav`

The results I get are not the same (and somehow a little weird..):

| Dataset    | FAD |
|----------------------|-------|
| Original Test Set    | 7.754 |
| Synthesized Test Set | 4.176 |
| Synthesized Test Set (model checkpoint = 0)| 42.48 |

The model checkpoint = 0 case makes sense, but seems like I have a lower score on synthesized test set. Would love to discuss with you about the details of how you calculated FAD, I am thinking if there might be some diff between our FAD implementation.

Another point to double check is: in `trainer.py`, is it that I should add `self.model = self.load_checkpoint(self.model,-1)` in the `mode=='test'` block to load the best model checkpoint for inference?

Thanks in advance!

**Comment by fcaspe** [comment]
Hello @gudgud96!

The procedure for computing the FAD looks ok, I also computed `ref_test.wav` and `synth_test.wav` each against the stats of the whole dataset present at the dir you indicated. One small detail is that I left out `AuSep_1_vn_18_Nocturne.wav` from the URMP violin dataset to use as a test case for continuous resynthesis. I did not train with data from this file and therefore this file is not considered on the FAD.

What looks strange to me is that the Original Test Set should always have a smaller FAD score than the resynthesized one, since `ref_test.wav` is made out of excerpts from the files located at `dataset/train/violin/*` (see `train.py` to check how I split data). This acts like kind of a sanity check for the whole test approach.

I used Google's original FAD repo located at [https://github.com/google-research/google-research/tree/master/frechet_audio_distance](url)

About loading the best model, if the training process finished correctly you will find a model called `state_best.pth`. You can resume from there with the option `resume_epoch = -1` in `config.yaml`.

Hope this helps!

Franco
